src/multivariate/composite.jl

```diff
@@ -0,0 +1,203 @@
+# Composite Multivariate distribution
+
+## Generic Composite multivariate distribution class
+
+module CompositeDistributions
+
+if !isdefined(:Distributions) using Distributions end
+
+import Base.length, Base.mean, Base.show
+import Distributions.params
+import Distributions.mean, Distributions.mode, Distributions.var, Distributions.cov
+import Distributions.entropy, Distributions.insupport
+import Distributions._logpdf, Distributions._pdf!
+import Distributions._rand!
+
+export AbstractCompositeContinuousDist, ContinuousMultivariateDistribution
+export GenericCompositeContinuousDist, CompositeDist
+export length, params, set_params!, mean, mode, var, cov, entropy
+export insupport, _logpdf, gradlogpdf, _rand!
+
+
+abstract AbstractCompositeContinuousDist <: ContinuousMultivariateDistribution
+#abstract AbstractCompositeDiscreteDist <: DiscreteMultivariateDistribution  # An idea, but not implemented yet
+
+immutable GenericCompositeContinuousDist <: AbstractCompositeContinuousDist
+    dist::Vector{ContinuousDistribution}
```

Comment:
You're right: my suggestion won't work. Given your goal of allowing different types in the vector, the only possible way to create a new type that will satisfy `isbits` (and therefore be amenable to stack allocation) is to use a tuple instead of a vector.